### PR TITLE
[AIRFLOW-2363] Fix return type bug in TaskHandler()._read()

### PR DESCRIPTION
**Pull request by KevinYang21**
### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW-2363/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2363
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Fix bug in S3TaskHandler()._read() so it always return the right tuple. Also added corresponding unit test to catch the error case.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
tests/utils/log/test_s3_task_handler.py:TestS3TaskHandler.test_read_when_s3_log_not_exist

- Performed end to end test on S3 setup, also tested by someone else end to end according to the conversation in https://issues.apache.org/jira/browse/AIRFLOW-2363

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by KevinYang21** [comment]
Hi @hamlinkn, do you think it is possible for you to test it in your infra end to end? It might take some extra work to have S3 set up on my side and I think you might benefit from a faster merge. Thank you!

**Comment by KevinYang21** [comment]
@aoen @Fokko PTAL

File: tests/utils/log/test_s3_task_handler.py [hunk 1]
```diff
@@ -117,6 +117,16 @@ def test_read(self):
              'Log line\n\n'], [{'end_of_log': True}])
         )
 
+    def test_read_when_s3_log_not_exist(self):
```
Comment (aoen) [inline tests/utils/log/test_s3_task_handler.py]: Nit: rename to test_read_when_s3_log_missing

File: tests/utils/log/test_s3_task_handler.py [hunk 2]
```diff
@@ -117,6 +117,16 @@ def test_read(self):
...
+        log, metadata = self.s3_task_handler.read(self.ti)
+
+        self.assertEqual(1, len(log))
+        self.assertEqual(len(log), len(metadata))
+        self.assertIn('*** Log file does not exist: local/log/location/1.log\n'
```
Comment (aoen) [inline tests/utils/log/test_s3_task_handler.py]: This seems a bit brittle (especially the port number), I think just checking "*** Log file does not exist" should be enough

File: airflow/bin/cli.py [hunk 3]
```diff
@@ -467,6 +467,7 @@ def run(args, dag=None):
     else:
         with redirect_stdout(ti.log, logging.INFO), redirect_stderr(ti.log, logging.WARN):
             _run(args, dag, ti)
+    logging.shutdown()
```
Comment (aoen) [inline airflow/bin/cli.py:470]: Why is this needed?

**Comment by KevinYang21** [comment]
@berislavlopac PTAL, this should address the bug in the [issue you opened](https://issues.apache.org/jira/browse/AIRFLOW-2374), feel free to test it on your side.

File: airflow/bin/cli.py (hunk 3 above)
Comment (KevinYang21) [reply airflow/bin/cli.py:470]: I removed this line in the Elasticsearch commit (because I saw this method registered in atexit()) and it causes NoneType error. My theory is that logging.shutdown() called in atexit() need settings.Session() to be defined but at the point it was called according to the order exit functions were registered, the orm was disposed. Explicitly calling it here is safe as all the task handler have logic prevent duplicate close() calls.

**Comment by berislavlopac** [comment]
I'll test on my setup, but can't do anything before Monday. :disappointed:

**Comment by Fokko** [comment]
@berislavlopac would be awesome if you can give it a spin. This should be in 1.10.0.

**Comment by berislavlopac** [comment]
I'll try to test it today.

**Comment by berislavlopac** [comment]
I'm trying to install it as `-e git://github.com/apache/incubator-airflow.git@refs/pull/3259/merge#egg=apache-airflow` but I'm getting the `unknown revision or path not in the working tree` error. Any suggestions how to set it up?

**Comment by Fokko** [comment]
I'm always using `pip install --upgrade git+https://github.com/yrqls21/incubator-airflow.git@kevin_yang_fix_s3_logging`

**Comment by berislavlopac** [comment]
Thank you, this worked! And I can report that it's working, the logs are now being sent to GCP! :+1:

**Review by Fokko — APPROVED** [review]
@berislavlopac Awesome, thanks!
